examples/ci-tests/i2c-master-rx/README.md

```diff
@@ -1,21 +1,64 @@
-# Hardware CI Tests
+# I2C Master RX Test
 
-## Run Python Test
+## Test Description
 
-`test.py` provides basic test that sends message from slave to master, and waits for master to send message containing the slave message. This is essentially a Rx/Tx test, but we are primarily focusing on the reception of master here. The message sent from master will be printed over a 12 second period. If message sent from master is the exact same from message sent from slave, the test will pass.
+`test.py` provides the following tests:
+
+### I2CMasterRxTest
+
+This receives message from the device-under-test as I2C Master from the RPi as I2C Slave.
+
+  1. The device-under-test configures itself as I2C master with address 0x40 and awaits button press.
+  1. The test harness presses a button (GPIO toggle) on device-under-test to indicate the device-under-test to prepare interaction and receive message as master.
+  1. The test harness issues an I2C write transaction of 16 bytes and awaits master to send back read data.
+  1. The test harness validates that the data received from device-under-test and sent back to test harness matches the expected string sent `"Hello I'm Slave"`.
```

Comment:
```suggestion
  1. The test harness prepares a buffer of 16 bytes that it will send in response to I2C read requests from the device-under-test.
  1. The test harness presses a button (GPIO toggle) on device-under-test to indicate the device-under-test should begin issuing transactions.
  1. The device-under-test issues repeated I2C read transactions of 16 bytes, which reads from the test harness.
  1. The device-under-test prints the data received over I2C on the console.
  1. The test harness validates that the data received from device-under-test and sent back to test harness matches the expected string sent `"Hello I'm Slave"`.
```

I re-arranged some of the language here to be a little clearer — I believe this should match what is actually happening, but please verify I got it correct.